File: C++/0118.cpp
#include <iostream>
#include <vector>

class Solution {
public:
    std::vector<std::vector<int>> generate(int numRows) {
        std::vector<std::vector<int>> res;
        if(numRows == 0) return res;
        std::vector<int> temp;
        temp.push_back(1);
        res.push_back(temp);

        for(int i = 1; i < numRows; i++){
            std::vector<int> temp;
            temp.push_back(1);
            for(int j = 1; j < i; j++){
                temp.push_back(res[i-1][j]+res[i-1][j-1]);
            }
            temp.push_back(1);
            res.push_back(temp);
        }
        return res;
    }
};

int main()
{
    Solution s;
    std::vector<std::vector<int>> res = s.generate(5);
    for(int i = 0; i < res.size(); i++)
    {
        for(int j = 0; j < res[i].size(); j++)
            std::cout << res[i][j] << " ";
        std::cout << std::endl;
    }
    return 0;
}

File: Python/0118.py
class Solution:
    def generate(self, numRows: int) -> list[list[int]]:
        ans = [[1]]
        for i in range(1, numRows):
            row = [1]
            for j in range(1, i):
                row.append(ans[i-1][j]+ans[i-1][j-1])
            row.append(1)
            ans.append(row)
        return ans


"""
Input: Integer
Output: List of lists of integers

First we need to create a list of lists of integers
We know that the first row is always 1

We can use a for loop to iterate through the rows
Each row will have a 1 at the beginning
Then we fill in the rest of the row with the sum of the previous row's elements
Each row will have a 1 at the end
We append each row to the list of lists

Time Complexity: O(n^2)
Space Complexity: O(n^2)
"""